ambry-store/src/main/java/com.github.ambry.store/StoreFindToken.java

```diff
@@ -128,11 +137,44 @@ static StoreFindToken fromBytes(DataInputStream stream, StoreKeyFactory factory)
             break;
           case JournalBased:
             Offset logOffset = Offset.fromBytes(stream);
-            storeFindToken = new StoreFindToken(logOffset, sessionIdUUID);
+            storeFindToken = new StoreFindToken(logOffset, sessionIdUUID, null);
+            break;
+          case IndexBased:
+            Offset indexSegmentStartOffset = Offset.fromBytes(stream);
+            storeFindToken =
+                new StoreFindToken(factory.getStoreKey(stream), indexSegmentStartOffset, sessionIdUUID, null);
+            break;
+          default:
+            throw new IllegalStateException("Unknown store find token type: " + type);
+        }
+        break;
+      case VERSION_2:
+        // read incarnationId
```

Comment:
it looks like an improvement to code readability can be made here. The only situation where `incarnationId` and `sessionId` are empty is when the token is uninitialized. I think we should read the type first and then read session and incarnation ids. That way, they are guaranteed to be non null. There might be a bit of repeated code in the switch case, but can you explore how it looks?